fxtest/lifecycle.go

```diff
@@ -66,34 +66,86 @@ func (t *panicT) FailNow() {
 	panic("test lifecycle failed")
 }
 
+// LifecycleOption modifies the behavior of the [Lifecycle]
+// when passed to [NewLifecycle].
+type LifecycleOption interface {
+	apply(*Lifecycle)
+}
+
+// EnforceTimeout will cause the [Lifecycle]'s Start and Stop methods
+// to return an error as soon as context expires,
+// regardless of whether specific hooks respect the timeout.
+func EnforceTimeout(enforce bool) LifecycleOption {
+	return &enforceTimeout{
+		enforce: enforce,
+	}
+}
+
+type enforceTimeout struct {
+	enforce bool
+}
+
+func (e *enforceTimeout) apply(lc *Lifecycle) {
+	lc.enforceTimeout = e.enforce
+}
+
+var _ LifecycleOption = (*enforceTimeout)(nil)
+
 // Lifecycle is a testing spy for fx.Lifecycle. It exposes Start and Stop
 // methods (and some test-specific helpers) so that unit tests can exercise
 // hooks.
 type Lifecycle struct {
 	t  TB
 	lc *lifecycle.Lifecycle
+
+	enforceTimeout bool
 }
 
 var _ fx.Lifecycle = (*Lifecycle)(nil)
 
 // NewLifecycle creates a new test lifecycle.
-func NewLifecycle(t TB) *Lifecycle {
+func NewLifecycle(t TB, opts ...LifecycleOption) *Lifecycle {
```

Comment:
Strictly speaking, this is considered a breaking change per Go compatibility guarantees.
Adding variadic options to a function that didn't previously accept them still changes the signature.
It breaks cases like passing a reference to the function, for example.

I'll defer to y'all about whether you'd like to take the risk, since passing `fastest.NewLifecycle` would be a pretty niche use.

Alternatively, we can add one of `NewLifecycleWith`, `NewLifecycleWithOptions`, or `NewStrictLifecycle` instead. (The last defaulting to enforceTimeout = true. In that case, we could drop the EnforceTimeout option completely, and instead just add `Option`s for future expansion.)

Reply:
I don't feel strongly about this; will defer to other maintainers.
I think this is fine as-is. Passing `fxtest.NewLifecycle` as a function parameter is a niche enough use case  that I expect zero breakage.

This isn't authoritative, but just searching public code on GitHub, there are no references to `fxtest.NewLifecycle` not followed by a `(`, so they're always invocations, and won't be broken.
https://github.com/search?q=%2Ffxtest%5C.NewLifecycle%5B%5E%5C%28%5D%2F+path%3A%2F.go%24%2F&type=code

Reply:
👍 @sywhang or @tchung1118 - do you have strong opinions on this? Otherwise I will merge this as-is.

Reply:
I think this is fine as-is, too.

Reply:
I don't have a strong opinion against this either but I do prefer making a slightly breaking change in this API rather than overloading the API surface with a variant of `NewLifecycle` given how niche the set of use cases that break from this.